GoogleGram/app/src/main/java/com/example/zghadyali/googlegram/FeedFragment.java

```diff
@@ -0,0 +1,122 @@
+        nextImage = (Button) rootView.findViewById(R.id.next);
+        nextImage.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                if (imagesLink.size() > 0) {
+                    image_index = image_index + 1;
+                    if (image_index >= imagesLink.size()) {
+                        image_index = 0;
+                    }
+                    web.loadUrl(imagesLink.get(image_index));
+                }
+            }
+        });
+        prevImage = (Button) rootView.findViewById(R.id.previous);
+        prevImage.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                if (imagesLink.size() > 0) {
+                    image_index = image_index - 1;
+                    if (image_index < 0) {
+                        image_index = imagesLink.size() - 1;
+                    }
+                    web.loadUrl(imagesLink.get(image_index));
+                }
+            }
+            });
+        //when the user clicks on the delete button, loads the WebView with the next image if there
+        //is one to see and deletes the link to the previous image from the database and the
+        //ArrayList
+        delete = (Button) rootView.findViewById(R.id.delete);
+        delete.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                if (imagesLink.size() > 1) {
+                    String link = imagesLink.get(image_index);
+                    DbHelper.deleteItemtoDB(link);
+                    imagesLink.remove(link);
+                    web.loadUrl(imagesLink.get(image_index));
+
+                }
+                else if (imagesLink.size() == 0){   //loads a blank WebView if deleted all images in
+                // feed
+                    web.loadUrl("about:blank");
+                }
+                else{   //deletes the only image in the ArrayList and loads a blank WebView
+                    image_index = 0;
+                    String link = imagesLink.get(image_index);
+                    DbHelper.deleteItemtoDB(link);
+                    imagesLink.remove(link);
+                    web.loadUrl("about:blank");
+                }
+            }
+        });
+        //when the user clicks the search button, initialize a new searchGoogle fragment and use the
+        //transitionToFragment method in MainActivity to load the searchGoogle fragment in the
+        //container
+        switchGoogle = (Button) rootView.findViewById(R.id.switch_google);
+        switchGoogle.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                SearchGoogleFragment searchGoogle = new SearchGoogleFragment();
+                ((MainActivity)getActivity()).transitionToFragment(searchGoogle);
+            }
+        });
```

Comment:
this could be set here, but please try to move your onClicklisteners out of your oncreate's